src/app/_components/CategorySelect.tsx

```diff
@@ -35,36 +37,65 @@ export function CategorySelect({
         touchTarget.offsetClassY,
       )}
     >
+      {children}
+    </ul>
+  )
+}
+
+export function CategorySelect({
+  categoryOption,
+  categorySettings,
+  categoryCounts,
+  onCategoryOptionChange,
+}: CategorySelectProps) {
+  return (
+    <CategoryWrapper>
       {categorySettings.map((option) => {
         const isSelected = categoryOption === option.id
 
         const countOrUndefined = categoryCounts
           ? categoryCounts[option.id] || 0
           : undefined
-        const hasCount = typeof countOrUndefined === 'number'
 
         return (
-          <li key={option.id}>
-            <button
-              className={clsx(
-                'text-pretty rounded-lg py-2 text-left font-bold focus:brand-outline hover:bg-brand-700',
-                hasCount && 'inline-flex items-baseline gap-2',
-                touchTarget.class,
-                {
-                  'bg-brand-700 text-brand-400': isSelected,
-                  'bg-transparent text-brand-300': !isSelected,
-                },
-              )}
-              onClick={() => onCategoryOptionChange(option.id)}
-            >
-              <span>{option.name}</span>
-              {hasCount && (
-                <span className="text-sm font-light">({countOrUndefined})</span>
-              )}
-            </button>
-          </li>
+          <CategoryListItem
+            key={option.id}
+            name={option.name}
+            isSelected={isSelected}
+            count={countOrUndefined}
+            handleClick={() => onCategoryOptionChange(option.id)}
+          />
         )
       })}
-    </ul>
+    </CategoryWrapper>
+  )
+}
+
+export function CategoryListItem({
+  name,
+  count,
+  isSelected,
+  handleClick,
```

Comment:
FYI, I get this error when I hover over `handleClick`. I'm not sure if it's relevant to us, just sharing.

_Props must be serializable for components in the "use client" entry file, "handleClick" is invalid.ts(71007)_

Reply:
Hm I couldn't recreate, what were the exact steps? Or could you record a video?

Reply:
Hmm not sure, I don't see it. You probably have some vs code extension I don't? `handleClick` is passed from a client component so not sure what could be an issue 🤔